src/fs/fs.c

```diff
@@ -40,6 +40,76 @@ LIST_HEAD(list, cfs_fs_node) fs_list = LIST_HEAD_INITIALIZER();
 /* global endpoint operations for cortxfs*/
 static const struct cfs_endpoint_ops *g_e_ops;
 
+int fs_node_init(struct cfs_fs_node *fs_node, struct namespace *ns,
+		 size_t ns_size)
+{
+	int rc = 0;
+
+	str256_t *fs_name = NULL;
+	ns_get_name(ns, &fs_name);
+
+	fs_node->cfs_fs.ns = calloc(1, ns_size);
+	if (fs_node->cfs_fs.ns == NULL) {
+		goto ns_alloc_fail;
+	}
+
+	memcpy(fs_node->cfs_fs.ns, ns, ns_size);
+
+	fs_node->cfs_fs.kvtree = calloc(1, sizeof(struct kvtree));
+	if (!fs_node->cfs_fs.kvtree) {
+		rc = -ENOMEM;
+		goto kvtree_alloc_fail;
+	}
+
+	rc = kvtree_init(fs_node->cfs_fs.ns, fs_node->cfs_fs.kvtree);
```

Comment:
You are just initializing the `kvtree_init` without creating `kvtree_create`. There is a minor difference between these two APIs, first one will initialize the kvtree pointer you have given and leave the kvs index opened where as second one will initialize the kvtree pointer we have given, dump the root node to kvstore and at the end it will close the kvs index.

As per old code flow, first we were creating the `kvtree_create` and then we were doing `kvtree_init`. So the doubt comes here is if `kvtree_create` is doing what `kvtree_init` is doing ( Initializing the kvtree pointer) then why there was extra `kvtree_init`, the reason for adding `kvtree_init` after `kvtree_create` was to keep name space index open as `kvtree_create` close the index when it is done. So there was some bug in which we found out all the fs indexes needs to be kept open and hence there is `kvtree_init` to keep th fs index open.

So you need to figure out what suits best here if we create the kvtree here or caller should create it and pass it by that way you will have kvtree created and you can avoid using calloc at line number 58.